console/src/components/Services/Data/TableModify/ModifyActions.js

```diff
@@ -164,6 +165,138 @@ const resetPrimaryKeys = () => ({
   type: RESET_PRIMARY_KEY,
 });
 
+export const saveComputedField = (
+  computedField,
```

Comment:
**Question**:

All these arguments are available in redux. Probably the only thing you need is their index. So not sure if we should pass all these arguments.

I was thinking that these ServerIO functions should have as less arguments as possible. Otherwise, as you iterate, things get pretty messy. Some of the functions in our code base have 8 arguments and all.

Tell me if you agree.

Reply:
I want to go in the opposite direction of this. I would want to make all ServerIO functions as pure functions which accept params explicitly and do not depend on the redux state. No of params is not a big issue if done logically. Doing this enures reusability and ability to move such code around easily. 

The more I think I about it, I believe redux state should only be used for holding common state and data and nothing else. Local state seems always enough for most rendering purposes.